modules/streamloader/src/main/scala/com/snowplowanalytics/snowplow/storage/bigquery/streamloader/Bigquery.scala

```diff
@@ -76,16 +99,35 @@ object Bigquery {
     )
   }
 
-  private def extractFailedAndSuccessfulRows(
+  private case class PartitionedResult(
+    failed: List[LoaderRow],
+    successful: List[LoaderRow],
+    stopped: List[LoaderRow],
+    reasonsAndLocations: Set[(String, String)]
+  )
+
+  private def partitionRowResults(
```

Comment:
This could be unit-tested. But the code is trivial enough, so I am not sure it is necessary.